### PR TITLE
Upgrade to closure v20171023

### PR DESCRIPTION
**Pull request by rkirov**
Files without goog.module or goog.provide, appear to report a new
synthetic provide 'module$src$...', which we now filter out in clutz.

### CODE REVIEW

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 1]
```diff
@@ -427,9 +427,20 @@ String produceDts(Depgraph depgraph) {
       String originalPath = compilerInput.getSourceFile().getOriginalPath();
       if (opts.skipEmitRegExp != null && Pattern.matches(opts.skipEmitRegExp, originalPath))
         continue;
-      transitiveProvides.addAll(compilerInput.getProvides());
+      Collection<String> inputProvides = compilerInput.getProvides();
+      Collection<String> filteredProvides = new ArrayList<>();
```
Comment (evmar) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: This seems unrelated to the change?

Comment (rkirov) [reply src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: It is related. Its in the commit description

Comment (rkirov) [reply src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: The new version of closure started reporting new synthetic provides. If we dont filter them out we create a bunch of garbage goog: module exports.

**Comment by rkirov** [comment]
Yes, simply upgrading closure, makes all goldens with empty files fail.
